packages/cdktf/test/json-to-hcl.test.ts

```diff
@@ -0,0 +1,975 @@
+// Copyright (c) HashiCorp, Inc
+// SPDX-License-Identifier: MPL-2.0
+import {
+  Testing,
+  TerraformStack,
+  TerraformLocal,
+  TerraformOutput,
+  TerraformVariable,
+  TerraformHclModule,
+  VariableType,
+} from "../lib";
+import {
+  OtherTestResource,
+  TestModule,
+  TestProvider,
+  TestResource,
+} from "./helper";
+import * as b from "../lib/backends";
+import { Construct } from "constructs";
+
+test("string local", async () => {
+  const app = Testing.app();
+  const stack = new TerraformStack(app, "test");
+
+  new TerraformLocal(stack, "greeting", "Hello, ${var.name}");
+
+  const hcl = Testing.synthHcl(stack);
+  expect(hcl).toMatchInlineSnapshot(`
+    "
+
+    locals {
+        greeting = "Hello, \${var.name}"
+    }"
+  `);
+});
+
+test("with provider alias", async () => {
+  const app = Testing.app();
+  const stack = new TerraformStack(app, "test");
+
+  new TestProvider(stack, "provider", {
+    accessKey: "1",
+  });
+
+  const provider = new TestProvider(stack, "aliased-provider", {
+    alias: "foo",
+    accessKey: "123",
+  });
+
+  new TestResource(stack, "test", {
+    name: "bar",
+    provider,
+  });
+
+  const hcl = Testing.synthHcl(stack);
+  expect(hcl).toMatchInlineSnapshot(`
+    "terraform {
+    required_providers {
+      test = {
+    version = "~> 2.0"
+    }
+    }
+
+
+    }
+
+    provider "test" {
+    access_key = 1
+    }
+
+    provider "test" {
+    access_key = 123
+    alias = "foo"
+    }
+    resource "test_resource" "test" {
+    name = "bar"
+    provider = "test.foo"
+    }"
+  `);
+});
+
+test("with formatting", async () => {
```

Comment:
:) I gave up on the formatting testing because it would need the `hcl-tools` to be imported, which `cdktf` does-not-like! Then I forgot to change the name